include/internal/refcount.h

```diff
@@ -57,9 +55,7 @@ static ossl_inline int CRYPTO_UP_REF(int *val, int *ret, void *lock)
 
 static ossl_inline int CRYPTO_DOWN_REF(int *val, int *ret, void *lock)
 {
-    *ret = __atomic_fetch_sub(val, 1, __ATOMIC_RELEASE) - 1;
-    if (*ret == 0)
-        __atomic_thread_fence(__ATOMIC_ACQUIRE);
+    *ret = __atomic_fetch_sub(val, 1, __ATOMIC_RELAXED) - 1;
```

Comment:
Why not `__atomic_sub_fetch` here which saves the `- 1` at the end?
Likewise for `atomic_fetch_sub_explicit` and the add equivalents.